tokens/chaptermember.inc

```diff
@@ -16,60 +16,49 @@
   }
 
   function chaptermember_get_tokens($cid, &$value, $onlyActive) {
-    $params = array(
-      'version' => 3,
-      'contact_id' => $cid,
-      'sequential' => 1,
-      'is_test' => 0,
-      'options' => array('limit' => 1, 'sort' => 'end_date DESC'),
-      'return' => array(CAGIS_CHAPTER),
-    );
-    if ($onlyActive == TRUE) {
-      $params['is_active'] = 1;
-      $params['active_only'] = 1;
-    }
-
     try {
-      $membership = civicrm_api3('membership', 'getsingle', $params);
-
-      // Get label for chapter
-      if (!empty($membership[CAGIS_CHAPTER])) {
-        $chapters = CRM_Core_OptionGroup::values('cagis_chapter');
-        $membership[CAGIS_CHAPTER] = $chapters[$membership[CAGIS_CHAPTER]];
-
-        // Now retrieve the Chapter Organization
-        $org = CRM_Core_DAO::singleValueQuery("SELECT id FROM civicrm_contact WHERE organization_name = %1", [1 => [$membership[CAGIS_CHAPTER], 'String']]);
+      $activeClause = $onlyActive ? ' AND m.is_active = 1' : '';
```

Comment:
Do we need to worry about statuses at all instead? I haven't checked what the config is like on CAGIS but wonder if there are any 'non current statuses' that are tagged as active

Reply:
yeah, I fixed it not pushed the commit yet as I am still testing code in staging.